sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -69,26 +69,35 @@ final class DefaultAndroidEventProcessor implements EventProcessor {
 
   private final @NotNull BuildInfoProvider buildInfoProvider;
   private final @NotNull RootChecker rootChecker;
+  private final @NotNull SentryAndroidOptions options;
 
   private final @NotNull ILogger logger;
 
   public DefaultAndroidEventProcessor(
       final @NotNull Context context,
       final @NotNull ILogger logger,
-      final @NotNull BuildInfoProvider buildInfoProvider) {
-    this(context, logger, buildInfoProvider, new RootChecker(context, buildInfoProvider, logger));
+      final @NotNull BuildInfoProvider buildInfoProvider,
+      final @NotNull SentryAndroidOptions options) {
+    this(
+        context,
+        logger,
```

Comment:
You could remove the `logger` ctor param now since `SentryAndroidOptions` provides that.